IMAGE_DEVELOP_BRANCHING_STRATEGY.md

```diff
@@ -0,0 +1,56 @@
+# Vanilla OS Image Development Branching Strategy
+
+Starting on July 24th, we are changing our branching strategy in Vanilla OS for image development. We hereby describe the design of the new branching system, its rules, and procedures for working with the following three branches: `vision`, `dev`, and `main`.
+
+## Merging vs. porting
+
+Before getting into the details of how the new branching strategy works, let's quickly clarify what 'merging' and 'porting' mean in this context.
+
+### Merging
+
+Merging is a process by which changes from one branch are integrated into another that shares the same code base. It ensures that the changes are included without affecting the structure and history of a project.
+
+### Porting
+
+While it is similar to merging, porting is the reimplementation from a feature or fix in one branch into another with dissimilar codebases.
+
+## Overview of Branches
+
+### Vision
+
+This `vision` branch serves as a base for experimental work, containing new features that could be backported into `dev`. It is a playground for new ideas and innovations that a developer can use for experimental purposes without hurting the development flow in `dev`.
+
+#### Merging Rules
+You should not do a direct merge from `vision` to `main`, and neither bring changes from `dev` into `vision`. This means that functionality developed on `vision` should be ported to `dev` instead of being merged directly.
+
+### Dev
+
+The `dev` branch is where active development happens, this includes work that is considered for the next stable release. Features and fixes are done, tested, prepared and put into this branch to be released as stable.
+
+#### Merging Rules
+Changes from `dev` should never be directly forwarded to `vision`. Features and fixes from `dev` shall always be merged to `main`, never ported.
+
+### Main
+
+The `main` branch contains code that is considered stable. Only tested and verified features and fixes are merged here.
+
+#### Merging Rules
```

Comment:
```suggestion
#### Merging Rules

You should not do a direct merge from `vision` to `main`, and neither bring changes from `dev` into `vision`. This means that functionality developed on `vision` should be ported to `dev` instead of being merged directly.

### Dev

The `dev` branch is where active development happens, this includes work that is considered for the next stable release. Features and fixes are done, tested, prepared and put into this branch to be released as stable.

#### Merging Rules

Changes from `dev` should never be directly forwarded to `vision`. Features and fixes from `dev` shall always be merged to `main`, never ported.

### Main

The `main` branch contains code that is considered stable. Only tested and verified features and fixes are merged here.

#### Merging Rules

```